### DISCUSSION TITLE
Limit "diffdisk" size

### DISCUSSION THREAD

**Discussion by Mrazator** [lima-vm/lima#828] [Q&A] [answered]
Is it possible to somehow limit the "diffdisk" size?

Using lima as docker host on M1 and the size of `~/.lima/docker/diffdisk` takes up to 49.01 GB.

- Is all of it really used?
- Is it possible to prune it somehow?
- What is is inside?

**Comment by jandubois** [comment, marked as answer]
You can configure the maximum size of the `diffdisk` when you create the Lima instance via the `disk` setting in `lima.yaml`:

https://github.com/lima-vm/lima/blob/2b8807175b322c3b46634e7f407b48e526eae2db/examples/default.yaml#L38-L40

The space is mostly used for container images; you can delete old images, but the space will not be returned to the host, it will only be reused for new images. You have to delete and recreate the instance to get the disk space back.

Issue #356 is about trimming the disk space (after pruning images), but it is not implemented, and I think it isn't clear if this is even possible on macOS.

**Reply by afbjorklund** [reply]
Does the trick to copy the VM image with `qemu-img convert` work on macOS ?

It needs to be shut down before attempting, though. It usually works on Linux

**Reply by afbjorklund** [reply]
> What is is inside?

I think it has some `docker system df` command ?

https://docs.docker.com/engine/reference/commandline/system_df/

**Comment by erlangparasu** [comment]
ls -laF -h
```
100G Oct 11 12:25 diffdisk
```
